conformance/lang/expresssions/literals/byte_array_literal.balt

```diff
@@ -0,0 +1,527 @@
+                                fabcd8912df`;
+
+    io:println(byteArray1); // @output [170,171,207,204,173,175,205,52,26,75,223,171,205,137,18,223]
+    io:println(byteArray1 == byteArray2); // @output true
+    io:println(byteArray1 == byteArray3); // @output true
+    io:println(byteArray1 == byteArray4); // @output true
+}
+
+Test-Case: output
+Description: Test inherent type of base16 byte array literal is byte[N],
+             where N is the number of bytes encoded by the Base16Literal.
+Fail-Issue: ballerina-platform/ballerina-lang#32542
+Labels: byte-array-literal, is-expr
+
+function init() {
+    io:println(base16 `` is byte[0]); // @output true
+    io:println(base16 `aaabcfccadafcd341a4bdfabcd8912df` is byte[16]); // @output true
+}
+
+Test-Case: output
+Description: Test static type of base16 byte array literal is byte[N],
+             where N is the number of bytes encoded by the Base16Literal.
+Fail-Issue: ballerina-platform/ballerina-lang#32542
+Labels: byte-array-literal
+
+function init() {
+    byte[16] byteArray1 = base16 `aaabcfccadafcd341a4bdfabcd8912df`;
+    byte[*] byteArray2 = base16 `aaabcfccadafcd341a4bdfabcd8912df`;
+    io:println(byteArray1); // @output [170,171,207,204,173,175,205,52,26,75,223,171,205,137,18,223]
+    io:println(byteArray1 == byteArray2); // @output true
+}
+
+Test-Case: output
+Description: Test assignment of base16 byte array literal.
+Labels: byte-array-literal, is-expr, int:Signed16, int:Signed32, int:Unsigned8, int:Unsigned16, int:Unsigned32
+
+function init() {
+    byte[] byteArray1 = base16 `aaabcfccadafcd341a4bdfabcd8912df`;
+    io:println(byteArray1 is byte[]); // @output true
+
+    int[] byteArray2 = base16 `aaabcfccadafcd341a4bdfabcd8912df`;
+    io:println(byteArray2 is int[]); // @output true
+
+    int:Unsigned8[] byteArray3 = base16 `aaabcfccadafcd341a4bdfabcd8912df`;
+    io:println(byteArray3 is int:Unsigned8[]); // @output true
+
+    int:Unsigned16[] byteArray4 = base16 `aaabcfccadafcd341a4bdfabcd8912df`;
+    io:println(byteArray4 is int:Unsigned16[]); // @output true
+
+    int:Unsigned32[] byteArray5 = base16 `aaabcfccadafcd341a4bdfabcd8912df`;
+    io:println(byteArray5 is int:Unsigned32[]); // @output true
+
+    int:Signed16[] byteArray6 = base16 `aaabcfccadafcd341a4bdfabcd8912df`;
+    io:println(byteArray6 is int:Signed16[]); // @output true
+
+    int:Signed32[] byteArray7 = base16 `aaabcfccadafcd341a4bdfabcd8912df`;
+    io:println(byteArray7 is int:Signed32[]); // @output true
+}
+
+Test-Case: error
+Description: Test invalid assignment of base16 byte array literal for built-in subtype of int.
+Labels: byte-array-literal, int:Signed8
+
+function init() {
+    int:Signed8[] _ = base16 `aabb12`; // @error incompatible types: expected 'int:Signed8[]', found 'byte[]'
+}
+
+Test-Case: error
+Description: Test invalid assignment of base16 byte array literal for user-defined subtype of int.
+Labels: byte-array-literal, module-type-defn, union-type, DecimalNumber
```

Comment:
We should probably also add singleton-type right?

Reply:
updated. thanks!